src/Libraries/DesignScriptBuiltin/Dictionary.cs

```diff
@@ -129,6 +129,13 @@ public override string ToString()
                 result.Append("}");
                 return result.ToString();
             }
+
+            //TODO this could cause unanticipated consequences -
```

Comment:
I think this is not safe - ideas for alternatives - maybe something like `TestTimeEquals`?

Reply:
what is the problem that required this solution?

Reply:
see above in description:
```
 serialization tests fail when comparing dictionaries now - I've added an Equals override for our Dictionary type - but I think this is quite risky as it can change runtime behavior - probably best to have some kind of test only equals method that we can call from serialization - appreciate some thoughts here. I think this is just a change in assert behavior in Nunit.
 ```

Reply:
Why not just have a test-only method in this class or in the test framework for now that we use only for tests instead of using Nunit.Assert?

Reply:
@aparajit-pratap I will give that a shot - since the issue only happens for dictionaries, I'd like to avoid writing custom comparers for all our types... will take a look if we can inject our own comparer into nunit.